From 6930f6e27d3b6e276d7c1cf2bf876e9f22a515a4 Mon Sep 17 00:00:00 2001
From: Octavian Purdila <octavian.purdila@nxp.com>
Date: Fri, 20 Jan 2017 13:40:50 +0200
Subject: [PATCH 1386/5242] MLK-13787 Revert "ARM: dts: imx6qdl-sabresd: Pass
 the hannstar panel compatible string"

commit  320170bfa1fc9d7556af37e7d9dc5907f99a1f42 from
https://source.codeaurora.org/external/imx/linux-imx.git

This reverts commit e0884948a48013bb2097f3e61c4f2122b22f6eb1.

The internal fbdev driver does not use the DRM infrastructure hence we
can't use the DRM simple panel driver, we have to stich with display
timings from device tree.

Signed-off-by: Octavian Purdila <octavian.purdila@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi |   30 +++++++++++++-----------------
 1 file changed, 13 insertions(+), 17 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index df034d7..d6bc0ae 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -245,7 +245,7 @@
 		status = "okay";
 	};
 
-	backlight_lvds: backlight-lvds {
+	backlight {
 		compatible = "pwm-backlight";
 		pwms = <&pwm1 0 5000000>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
@@ -280,17 +280,6 @@
 		status = "okay";
 	};
 
-	panel {
-		compatible = "hannstar,hsd100pxn1";
-		backlight = <&backlight_lvds>;
-
-		port {
-			panel_in: endpoint {
-				remote-endpoint = <&lvds0_out>;
-			};
-		};
-	};
-
 	v4l2_out {
 		compatible = "fsl,mxc_v4l2_output";
 		status = "okay";
@@ -1016,11 +1005,18 @@
 		fsl,data-width = <18>;
 		status = "okay";
 
-		port@4 {
-			reg = <4>;
-
-			lvds0_out: endpoint {
-				remote-endpoint = <&panel_in>;
+		display-timings {
+			native-mode = <&timing0>;
+			timing0: hsd100pxn1 {
+				clock-frequency = <65000000>;
+				hactive = <1024>;
+				vactive = <768>;
+				hback-porch = <220>;
+				hfront-porch = <40>;
+				vback-porch = <21>;
+				vfront-porch = <7>;
+				hsync-len = <60>;
+				vsync-len = <10>;
 			};
 		};
 	};
-- 
1.7.9.5

